### PR TITLE
Makefile.template: removed the version number from library names

### PR DESCRIPTION
**Pull request by emmenlau**
Required for MINGW or MSYS2 compilation

### CODE REVIEW

**Comment by jtattermusch** [comment]
Please elaborate on "Required for MINGW or MSYS2 compilation".

**Comment by emmenlau** [comment]
As far as I could see, the Makefile does not handle the variables SHARED_VERSION_CPP and SHARED_VERSION_CSHARP correctly on MINGW target. The compile eventually fails because it expects a dependency with '-1' suffix that is not built. This might be true for other platforms too, but since no other platform sets those variables, the error might be occluded.

I do not know if MINGW or MSYS2 users would prefer dll names with the numbered suffix. My personal impression is that the number-prefix is more common in Visual Studio, and less common in MINGW / MSYS2. Therefore I removed the number-suffix, consistent with Linux and Darwin targets. This is a personal and certainly biased decision, though!

File: templates/Makefile.template [hunk 1]
```diff
@@ -405,8 +405,8 @@
   SHARED_EXT_CSHARP = dll
   SHARED_PREFIX =
   SHARED_VERSION_CORE = -${settings.core_version.major}
```
Comment (jtattermusch) [inline templates/Makefile.template]: Why are we keeping the content of SHARED_VERSION_CORE but removing SHARED_VERSION_CPP and SHARED_VERSION_CSHARP?

**Comment by jtattermusch** [comment]
this is ok to test

**Comment by jtattermusch** [comment]
Superseded by #9342

**Pull request closed without merging**